Improve mobile login experience: Add language dropdown selector and remove excessive headroom

ljpendergrass opened the pull request:
Here's a quick hello-world contribution that converts the overflowing language menu on the login screen to a Bootstrap 4.6 dropdown, killing a stray `<style>` tag in the process, and updates the mobile experience to not have so much wasted space.

**From:**
<img width="376" alt="Screenshot 2024-03-06 at 1 33 30 PM" src="https://github.com/Gig-o-Matic/GO3/assets/3155832/231b448d-629a-4095-b71e-892ef8c0228c">

**To:**

<img width="384" alt="Screenshot 2024-03-06 at 1 47 24 PM" src="https://github.com/Gig-o-Matic/GO3/assets/3155832/37403cc1-0c51-4d64-a5a4-702316d81e11">
<img width="375" alt="Screenshot 2024-03-06 at 1 33 42 PM" src="https://github.com/Gig-o-Matic/GO3/assets/3155832/8bb7680b-cd23-4571-9681-2fe1a865f497">

The language `form`s and `inputs` still function. The dropdown's Bootstrappiness allows it to avoid the bottom of the screen when too close to the bottom of the screen. It also retains being responsive to keyboard input.

**Potential improvements:** The word "Language" could be translated, but I don't know enough about Django translations right now. If that would be a desired improvement I can invest the time this week to wire that up properly in this PR.

--------------

**The post-amble:** Greetings! I'm Logan, Long time gig-o-er, first time contributor. When Tyler F brought you up in our band (TMB) discord I got super excited by this revitalization effort. I only have a few nights-and-weekends level experience with Python but am a pixel-pusher for WordPress frontends and beyond in my day job.

I'll be trying to contribute to issues close to the frontend, though I am eager to get some real world experience with Python. Would love to hear about ways I can contribute in the realm of usability, long term vision for the frontend, and any needed facelifts. Some other possibilities for front end updates in the future I notice include:

- FontAwesome integration commented out? (I added an SVG icon in this change)
- Upgrade Bootstrap to 5.x or otherwise?

Thanks and let me know if there's anything I've missed, or any checklist I should refer to for PRs.

aaronoppenheimer commented:
Hi Logan! Welcome aboard. That change looks great.

A couple of things:
- Gig-o is using font-awesome - that commented out line in go3base.html is followed by one that actually does the work. The commented out one can go!
- So you should be able to replace that VSG with <i class="fa-solid fa-language"></i>
- But that didn't work...because it turns out the "kit" I had set up with FontAwesome 5 years ago didn't have that icon. I updated the kit to the latest version and now it should work.
- I like FontAwesome a lot, but the gig-o is on the free tier and they send me warnings sometimes that we've gone over the pageview limit. I ignore them and nobody has ever complained that they can't get their icons to work, but...still, seems like sometime someone should figure out.
- Translations are easy but I still owe some documentation. It uses the standard django translation system - I'll put some instructions in the wiki and will let you know. But what would you translate the word "Language" to?

aaronoppenheimer commented:
I added a page on internationalization to the wiki, so if you do get inspired to change the text, you can see how to do it there.

ljpendergrass commented:
Awesome, thanks! I just figure it would be tidy to have "language" localized along with the rest of the login page. I added the translations; it touched a lot of files in the process, but that seems to be expected. Let me know if touching this many files for a small change is not desired and we can move it out.

For posterity, I had called out the commented out FontAwesome link in the head, but once I noticed the FA5 implementation, I added a working FA5 Language icon and removed my comment about it. I removed it with another commit, lemme know if you think this is a good PR to do that in.

Copy that about the free tier. There is an official Django integration for FA that might allow to the project to serve FA assets, but I don't think that's worth spending energy on in the short term.

aaronoppenheimer commented:
All seems good to me!
